WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift

```diff
@@ -485,6 +490,25 @@ private extension StoreStatsV4PeriodViewController {
     ///
     /// - Parameter selectedIndex: the index of interval data for the bar chart. Nil if no bar is selected.
     func updateUI(selectedBarIndex selectedIndex: Int?) {
+
+        if timeRange.isCustomTimeRange {
+            // For WordPress.com sites or fully Jetpack-connected sites, toggle the visibility of visit data based on whether
+            // a custom range bar value is selected.
+            // If selected, display the visit data, as it's accurate for individual values.
+            // If deselected, hide the data, as it's inaccurate for the entire range.
+            // This doesn't apply to Jetpack CP sites or non-Jetpack self-hosted sites, as they don't have visitor data.
+
+            guard let site = stores.sessionManager.defaultSite else { return }
+
+            if site.isWordPressComStore || (site.isJetpackConnected && !site.isJetpackCPConnected) {
+                if selectedIndex != nil && siteVisitStatsMode == .redactedDueToCustomRange {
+                    siteVisitStatsMode = .default
+                } else if selectedIndex == nil && siteVisitStatsMode == .default {
+                    siteVisitStatsMode = .redactedDueToCustomRange
+                }
+            }
```

Comment:
This can be simplified for better readability:

```suggestion
            if site.isJetpackConnected && site.isJetpackThePluginInstalled {
                siteVisitStatsMode = selectedIndex != nil ? .default : .redactedDueToCustomRange
            }
```